ext/test/key_provider/key_provider.c

```diff
@@ -0,0 +1,443 @@
+}
+
+/* Configuration parsing helpers */
+
+static int
+configure_int(const char *param, const WT_CONFIG_ITEM *k, const WT_CONFIG_ITEM *v, int *dest)
+{
+    if (strncmp(param, k->str, k->len) == 0 && k->len == strlen(param) && v->len > 0 &&
+      v->type == WT_CONFIG_ITEM_NUM) {
+        *dest = (int)v->val;
+
+        return (0);
+    }
+
+    return (EINVAL);
+}
+
+/*
+ * kp_configure --
+ *     Parse configuration options for the key provider.
+ */
+static int
+kp_configure(KEY_PROVIDER *kp, WT_CONFIG_ARG *config)
+{
+    WT_EXTENSION_API *wtext = kp->wtext;
+    WT_CONFIG_PARSER *config_parser = NULL;
+    WT_CONFIG_ITEM k = {0}, v = {0};
+    int ret = 0;
+
+    if ((ret = wtext->config_parser_open_arg(wtext, NULL, config, &config_parser)) != 0) {
+        LOG_ERROR(kp, NULL, "WT_EXTENSION_API.config_parser_open_arg: error: %d (%s)", ret,
+          wtext->strerror(wtext, NULL, ret));
+        goto err;
+    }
+
+    /* Parse configuration key-value pairs */
+    while ((ret = config_parser->next(config_parser, &k, &v)) == 0) {
+        if (configure_int("verbose", &k, &v, &kp->verbose) == 0)
+            continue;
+        else if (configure_int("key_expires", &k, &v, &kp->key_expires) == 0)
+            continue;
+
+        LOG_ERROR(kp, NULL, "WT_CONFIG_PARSER.next: unexpected configuration: %.*s=%.*s",
+          (int)k.len, k.str, (int)v.len, v.str);
+        ret = EINVAL;
+        goto err;
+    }
+
+    if (ret != WT_NOTFOUND) {
+        LOG_ERROR(kp, NULL, "WT_CONFIG_PARSER.next: error: %d (%s)", ret,
+          wtext->strerror(wtext, NULL, ret));
+        goto err;
+    }
+
+    ret = config_parser->close(config_parser);
+    config_parser = NULL;
+    if (ret != 0) {
+        LOG_ERROR(kp, NULL, "WT_CONFIG_PARSER.close: error: %d (%s)", ret,
+          wtext->strerror(wtext, NULL, ret));
+        goto err;
+    }
+
+    return (0);
+
+err:
+    if (config_parser != NULL)
+        (void)config_parser->close(config_parser);
+
+    return (ret);
+}
+
+/*
+ * wiredtiger_extension_init --
+ *     WiredTiger test key provider extension initialization.
+ */
+int
+wiredtiger_extension_init(WT_CONNECTION *conn, WT_CONFIG_ARG *config)
+{
+    WT_EXTENSION_API *wtext = conn->get_extension_api(conn);
+
+    /* Allocate the key provider structure */
+    KEY_PROVIDER *kp;
+    if ((kp = calloc(1, sizeof(KEY_PROVIDER))) == NULL) {
+        wtext->err_printf(wtext, NULL, "%s: %s", __func__, wtext->strerror(wtext, NULL, ENOMEM));
+        return (ENOMEM);
+    }
+
+    kp->wtext = wtext;
+    kp->verbose = WT_VERBOSE_INFO; /* Default verbosity level */
+    kp->key_expires = 43200;       /* Default: 12 hours = 43200 seconds */
+    kp->state.key_time = 0;        /* Key time is 0 until first get_key call */
+
+    int ret = 0;
+    WT_KEY_PROVIDER *wtkp = (WT_KEY_PROVIDER *)kp;
+
+    /* Parse configuration options */
+    if ((ret = kp_configure(kp, config)) != 0)
+        goto err;
+
+    if (kp->key_expires == KEY_EXPIRES_ALWAYS) {
```

Comment:
I would rather us set a DEFAULT_KEY all the time, and then let it be overriden during `load_key`. This would eliminate the dependency on DEFAULT_KEYS being on KEY_EXPIRES_ALWAYS